cloudformation/custom_resource/cluster.yaml

```diff
@@ -304,6 +332,76 @@ Resources:
       Layers:
         - !Ref PclusterLayer
 
+  CleanupS3bucketFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Timeout: 60
+      Code:
+        ZipFile: !Sub |
+          import boto3
+          import logging
+          import sys
+          from botocore.config import Config
+          logger = logging.getLogger()
+          logger.setLevel(logging.INFO)
+          
+          crhelper_path = "/opt/python/pcluster/resources/custom_resources/custom_resources_code"
+          sys.path.insert(0, crhelper_path)
+          from crhelper import CfnResource
+          helper = CfnResource()
+          
+          def _delete_s3_artifacts(event):
+              """
+              Delete artifacts under the directory that is passed in.
+
+              It exits gracefully if directory does not exist.
+              :param bucket_name: bucket containing cluster artifacts
+              """
+              bucket_name = event["ResourceProperties"]["S3Bucket"]
+              try:
+                  if bucket_name != "NONE":
+                      bucket = boto3.resource("s3", config=Config(retries={"max_attempts": 60})).Bucket(bucket_name)
+                      logger.info("Cluster S3 artifact in %s deletion: STARTED", bucket_name)
+                      bucket.objects.all().delete()
+                      bucket.object_versions.delete()
+                      logger.info("Cluster S3 artifact in %s deletion: COMPLETED", bucket_name)
+              except boto3.client("s3").exceptions.NoSuchBucket as ex:
+                  logger.warning("S3 bucket %s not found. Bucket was probably manually deleted.", bucket_name)
+                  logger.warning(ex, exc_info=True)
+              except Exception as e:
+                  logger.error(
+                      "Failed when deleting cluster S3 artifact in %s with error %s", bucket_name, e
+                  )
+                  raise
+
+          @helper.create
+          @helper.update
+          def no_op(_, __):
+              pass
+
+          @helper.delete
+          def delete(event, _):
+              _delete_s3_artifacts(event)
+
+
+          def handler(event, context):
+              helper(event, context)
```

Comment:
What are the event pass in here? Are we risk to pass the cluster config json into the event here?

Reply:
In this template
``` 
 ServiceToken:
    Description: Lambda for managing PCluster Resources
    Value: !GetAtt PclusterCfnFunction.Arn
```
So the cluster configuration is only passed in to the PclusterCfnFunction, not this function